alternator/streams.cc

```diff
@@ -519,14 +519,26 @@ future<executor::request_return_type> executor::describe_stream(client_state& cl
         // stored order, which is not neccesarily token order,
         // so the finding of "closest" token boundary (using upper bound)
         // could give somewhat weird results.
-        static auto cmp = [](const cdc::stream_id& id1, const cdc::stream_id& id2) {
+        static auto token_cmp = [](const cdc::stream_id& id1, const cdc::stream_id& id2) {
             return id1.token() < id2.token();
         };
 
+        // normal bytes compare is string_traits<int8_t>::compare.
+        // thus bytes 0x8000 is less than 0x0000. By doing unsigned
+        // compare instead we inadvertently will sort in string lexical.
+        using lexical_view = std::basic_string_view<uint8_t>;
+        static auto make_lexical_view = [](const bytes& b) {
+            return lexical_view(reinterpret_cast<const uint8_t*>(b.data()), b.size());
+        };
+        static auto id_cmp = [](const cdc::stream_id& id1, const cdc::stream_id& id2) {                    
+            return make_lexical_view(id1.to_bytes()) < make_lexical_view(id2.to_bytes());
+        };
```

Comment:
All of this looks correct, but in bytes.hh we have a function `compare_unsigned()` to do exactly the comparison you need (you only need "<" so you need `compare_unsigned(..,...) < 0`).
We had a bug like this in Alternator in the past - see commit f6b1f45d696acc0fff5922e0bd3c31e7c6348894
That being said, I'm ok with your re-implementation, it seems simple enough (I don't know about the performance). Your call.

Reply:
Missed that one. Updated to use it.